File: content/2-sorting/1-bubble-sort/js/bubbleSort.js
function bubbleSort(arr) {
  let len = arr.length;
  let swapped;

  do {
    swapped = false;
    for (let i = 0; i < len - 1; i++) {
      if (arr[i] > arr[i + 1]) {
        [arr[i], arr[i + 1]] = [arr[i + 1], arr[i]];
        swapped = true;
      }
    }
  } while (swapped);

  return arr;
}

File: content/2-sorting/1-bubble-sort/rb/bubble_sort.rb
def bubble_sort(arr)
  n = arr.length
  (n-1).times do |i|
    (n-i-1).times do |j|
      if arr[j] > arr[j+1]
        arr[j], arr[j+1] = arr[j+1], arr[j]
      end
    end
  end
  arr
end